src/peep/peep.c

```diff
@@ -3421,7 +3478,787 @@ static void peep_update_fixing(int steps, rct_peep* peep){
+			return;
+		}
+	}
+
+	//  loc_6C14A2:
+
+	uint16 cx = ax >> 8;
+	ax &= 0x00FF;
+
+	ax = ax << 5;
+	cx = cx << 5;
+
+	ax += 16;
+	cx += 10;
+
+	rct_xy16 direction = word_981D6C[peep->var_78];
+
+
+	ax += direction.x * 20;
+	cx += direction.y * 20;
+
+	peep->destination_x = ax;
+	peep->destination_y = cx;
+	peep->destination_tolerence = 2;
+
+	loc_6C14E5(steps, peep, ride);
+}
+
+static void loc_6C14E5(int steps, rct_peep *peep, rct_ride *ride) {
+	invalidate_sprite_2((rct_sprite *) peep);
+	int out = peep_update_action(0, 0, 0, peep);// TODO: arguments
+	if (out == 0) {
+		// jnb
+		loc_6C1653(steps, peep, ride);
+		return;
+	}
+
+	sprite_move(unknown, unkown, dx, peep); // TODO: arguments
+	invalidate_sprite_2((rct_sprite *) peep);
+}
+
+void loc_6C1504(int steps, rct_peep *peep, rct_ride *ride) {
+	if (peep->state == PEEP_STATE_INSPECTING) {
+		sub_6B7588(peep, peep->current_ride);
+
+		peep->staff_rides_inspected++;
+		peep->window_invalidate_flags |= 10;
+		loc_6C1653(steps, peep, ride);
+		return;
+	}
+
+	// loc_6C1524:
+	peep->staff_rides_fixed++;
+	peep->window_invalidate_flags |= 10;
+
+	peep->sprite_direction = peep->var_78 << 3;
+	peep->action = PEEP_ACTION_STAFF_ANSWER_CALL_2;
+	peep->action_frame = 0;
+	peep->action_sprite_image_offset = 0;
+
+	sub_693B58(peep);
+	invalidate_sprite_2((rct_sprite *) peep);
+
+	loc_6C1559(steps, peep, ride);
+}
+
+static void loc_6C1559(int steps, rct_peep *peep, rct_ride *ride) {
+	if (peep->action != 0xFF) {
+		peep_update_action(0, 0, 0, peep); // TODO: arguments
+		return;
+	}
+
+	// loc_6C156F:
+	ride_fix_breakdown(peep->current_ride, 0); // TODO: What should be the reliabilityIncreaseFactor?
+
+	loc_6C1653(steps, peep, ride);
+}
+
+void loc_6C157E(int steps, rct_peep *peep, rct_ride *ride) {
+	uint32 ebx = peep->current_ride_station;
+
+	uint16 ax = ride->exits[ebx];
+	if (ax == 0xFFFF) {
+		ax = ride->entrances[ebx];
+
+		if (ax == 0xFFFF) {
+			// loc_6C1644:
+			peep_decrement_num_riders(peep);
+			peep->state = 0;
+			peep_window_state_update(peep);
+			return;
+		}
+	}
+
+	// loc_6C15AC
+
+	uint16 cx = ax >> 8;
+	ax = ax & 0x00FF;
+	ax = ax << 5;
+	cx = cx << 5;
+
+	ebx = peep->var_78;
+
+	ax += 16;
+	cx += 16;
+
+	rct_xy16 ebx_direction = word_981D6C[ebx];
+	ax -= ebx_direction.x * 19;
+	cx -= ebx_direction.y * 19;
+
+	peep->destination_x = ax;
+	peep->destination_y = cx;
+	peep->destination_tolerence = 2;
+
+	loc_6C15EF(steps, peep, ride);
+}
+
+static void loc_6C15EF(int steps, rct_peep *peep, rct_ride *ride) {
+	invalidate_sprite_2(peep);
+	sint16 distance = 0;
+	int out = peep_update_action(0, 0, &distance, peep); // TODO: arguments
+	if (out == 0) {
+		// jnb
+		// loc_6C1644:
+		return;
+	}
+
+	uint32 ebx = peep->current_ride_station;
+	uint16 dx = ride->station_heights[ebx];
+	dx = dx << 3;
+
+	if (distance == 16) {
+		sprite_move(); // TODO: arguments
+		invalidate_sprite_2((rct_sprite *) peep);
+		return;
+	}
+
+	// TODO: Figure out what happens at 0x6C1626
```

Comment:
It adds a bit onto z depending on ride type and then does the sprite move, invalidate.